test/test_utils/security.py

```diff
@@ -187,3 +262,280 @@ def get_ecr_scan_allowlist_path(image_uri):
+    :param image_allowlist: ScanVulnerabilityList, Vulnerabities that are present in the respective allowlist in the DLC git repo.
+    :param ecr_image_vulnerability_list: ScanVulnerabilityList, Vulnerabities recently detected WITHOUT running apt-upgrade on the originally released image.
+    :param upgraded_image_vulnerability_list: ScanVulnerabilityList, Vulnerabilites exisiting in the image WITH apt-upgrade run on it.
+    :param s3_bucket_for_storage: s3 name of the bucket that would be used for saving all the important data that needs to be stored during failure routine.
+    :return: dict, a dictionary consisting of the entire summary of the steps run within this method.
+    """
+    s3_filename_for_allowlist = save_scan_vulnerability_list_object_to_s3_in_json_format(
+        image, upgraded_image_vulnerability_list, "allowlist", s3_bucket_for_storage
+    )
+    s3_filename_for_current_image_ecr_scan_list = save_scan_vulnerability_list_object_to_s3_in_json_format(
+        image, ecr_image_vulnerability_list, "current-ecr-scanlist", s3_bucket_for_storage
+    )
+    original_filepath_for_allowlist = get_ecr_scan_allowlist_path(image)
+    edited_files = [{"s3_filename": s3_filename_for_allowlist, "github_filepath": original_filepath_for_allowlist}]
+    vulnerabilities_fixable_by_upgrade = get_vulnerabilites_fixable_by_upgrade(
+        image_allowlist, ecr_image_vulnerability_list, upgraded_image_vulnerability_list
+    )
+    newly_found_non_fixable_vulnerabilites = upgraded_image_vulnerability_list - image_allowlist
+    fixable_list = {}
+    if vulnerabilities_fixable_by_upgrade:
+        fixable_list = vulnerabilities_fixable_by_upgrade.vulnerability_list
+    apt_upgrade_list_filename = f"apt-upgrade-list-{test_utils.get_processor_from_image_uri(image)}.txt"
+    s3_filename_for_apt_upgrade_list = s3_filename_for_allowlist.replace("allowlist.json", apt_upgrade_list_filename)
+    original_filepath_for_apt_upgrade_list = "/".join(
+        original_filepath_for_allowlist.split("/")[:-1] + [apt_upgrade_list_filename]
+    )
+    new_package_list = fixable_list if isinstance(fixable_list, list) else list(fixable_list.keys())
+    create_and_save_package_list_to_s3(
+        original_filepath_for_apt_upgrade_list,
+        new_package_list,
+        s3_filename_for_apt_upgrade_list,
+        s3_bucket_for_storage,
+    )
+    edited_files.append(
+        {"s3_filename": s3_filename_for_apt_upgrade_list, "github_filepath": original_filepath_for_apt_upgrade_list}
+    )
+    newly_found_non_fixable_list = {}
+    if newly_found_non_fixable_vulnerabilites:
+        newly_found_non_fixable_list = newly_found_non_fixable_vulnerabilites.vulnerability_list
+    message_body = {
+        "edited_files": edited_files,
+        "fixable_vulnerabilities": fixable_list,
+        "non_fixable_vulnerabilities": newly_found_non_fixable_list,
+    }
+    _invoke_lambda(function_name="trshanta-ECR-AS", payload_dict=message_body)
```

Comment:
Please deploy the final lambda, and add its name as a global constant in `test_utils.security` or something.

Reply:
Done, thanks!!